crates/goose-cli/src/recipes/search_recipe.rs

```diff
@@ -1,31 +1,71 @@
 use anyhow::{anyhow, Result};
 use goose::config::Config;
-use std::fs;
 use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+use crate::recipes::recipe::RECIPE_FILE_EXTENSIONS;
 
 use super::github_recipe::{retrieve_recipe_from_github, GOOSE_RECIPE_GITHUB_REPO_CONFIG_KEY};
 
+const GOOSE_RECIPE_PATH_ENV_VAR: &str = "GOOSE_RECIPE_PATH";
+
 pub fn retrieve_recipe_file(recipe_name: &str) -> Result<(String, PathBuf)> {
-    // If recipe_name ends with yaml or json, treat it as a direct path
-    if recipe_name.ends_with(".yaml") || recipe_name.ends_with(".json") {
+    // If recipe_name ends with yaml or json, treat it as a direct file path
+    if RECIPE_FILE_EXTENSIONS
+        .iter()
+        .any(|ext| recipe_name.ends_with(&format!(".{}", ext)))
+    {
         let path = PathBuf::from(recipe_name);
         return read_recipe_file(path);
     }
-
-    // First check current directory
-    let current_dir = std::env::current_dir()?;
-    if let Ok((content, recipe_parent_dir)) = read_recipe_in_dir(&current_dir, recipe_name) {
-        return Ok((content, recipe_parent_dir));
-    }
-    read_recipe_in_dir(&current_dir, recipe_name).or_else(|e| {
+    retrieve_recipe_from_local_path(recipe_name).or_else(|e| {
         if let Some(recipe_repo_full_name) = configured_github_recipe_repo() {
+            println!("{}", e);
             retrieve_recipe_from_github(recipe_name, &recipe_repo_full_name)
         } else {
             Err(e)
         }
     })
 }
 
+fn retrieve_recipe_from_local_path(recipe_name: &str) -> Result<(String, PathBuf)> {
```

Comment:
This method seems very noisy now; I would expect it to be just search for the recipe first in the current directory and if the environment variable is set, also search those paths, but no need I would think to tell the user if the variable is not set or where you searched - just if it can't be found, report that.

The other thing is that I think you want to search for the recipe directly not for the recipe_name/recipe.json|yaml. I know this is different from how we do it with the git integration, but I think that makes sense.

That would simplify the whole thing to like in python:

for path in [".", *env['GOOSER_RECIPE_PATH'].split(":")]:
   for ext in EXTENSIONS:
      if os.path.exists(path + "/" + recipe_name + "." + ext): etc